rs/nns/sns-wasm/src/sns_wasm.rs

```diff
@@ -3266,10 +3278,10 @@ mod test {
 
         let subnet_id = subnet_test_id(1);
 
-        let root_id = canister_test_id(1);
-        let governance_id = canister_test_id(2);
-        let ledger_id = canister_test_id(3);
-        let swap_id = canister_test_id(4);
+        let governance_id = canister_test_id(1);
+        let ledger_id = canister_test_id(2);
+        let swap_id = canister_test_id(3);
+        let index_id = canister_test_id(4);
```

Comment:
Why did the order of canisters change? @anchpop

Reply:
There was a small bug where they were not deleted in the same order they were created. You can see that in the previous state before the PR (and in the current state) they were created in the following order

```
        Ok(SnsCanisterIds {
            root: Some(canisters_created.remove(0).get()),
            governance: Some(canisters_created.remove(0).get()),
            ledger: Some(canisters_created.remove(0).get()),
            swap: Some(canisters_created.remove(0).get()),
            index: Some(canisters_created.remove(0).get()),
        })
```

And a canister creation failure occurs for the first created canister in this test, e.g. root. But the function was previously reporting that the last canister (index) could not be created rather than the first. I assume that someone wrote the code and then evaluated it to determine what to test, and didn't notice this discrepancy. When I fixed the bug in the create_sns_canisters function, the test had to be updated also so that the behavior expected by the test was the correct behavior